the_wizz/kdtree_utils.py

```diff
@@ -204,3 +204,175 @@ def get_mean_array(self):
+        """
+        ras_rad = np.deg2rad(RA)
+        decs_rad = np.deg2rad(DEC)
+        try:
+            pos_sphere = np.empty((len(RA), 3))
+        except TypeError:
+            pos_sphere = np.empty((1, 3))
+        cos_decs = np.cos(decs_rad)
+        pos_sphere[:, 0] = np.cos(ras_rad) * cos_decs
+        pos_sphere[:, 1] = np.sin(ras_rad) * cos_decs
+        pos_sphere[:, 2] = np.sin(decs_rad)
+        return np.squeeze(pos_sphere)
+
+    @staticmethod
+    def _distance_sky2sphere(dist_sky):
+        """
+        _distance_sky2sphere(dist_sky)
+
+        Converts angular separation in celestial coordinates to the
+        Euclidean distance in (x, y, z) space.
+
+        Parameters
+        ----------
+        dist_sky : float or array_like
+            Single or list of separations in celestial coordinates.
+
+        Returns
+        -------
+        dist_sphere : float or array_like
+            Celestial separation converted to (x, y, z) Euclidean distance.
+        """
+        dist_sky_rad = np.deg2rad(dist_sky)
+        dist_sphere = np.sqrt(2.0 - 2.0 * np.cos(dist_sky_rad))
+        return dist_sphere
+
+    @staticmethod
+    def _distance_sphere2sky(dist_sphere):
+        """
+        _distance_sphere2sky(dist_sphere)
+
+        Converts Euclidean distance in (x, y, z) space to angular separation in
+        celestial coordinates.
+
+        Parameters
+        ----------
+        dist_sphere : float or array_like
+            Single or list of Euclidean distances in (x, y, z) space.
+
+        Returns
+        -------
+        dist_sky : float or array_like
+            Euclidean distance converted to celestial angular separation.
+        """
+        dist_sky_rad = np.arccos(1.0 - dist_sphere**2 / 2.0)
+        dist_sky = np.rad2deg(dist_sky_rad)
+        return dist_sky
+
+    def query_radius(self, RA, DEC, r):
+        """
+        query_radius(RA, DEC, r)
+
+        Find all data points within an angular aperture r around a reference
+        point with coordiantes (RA, DEC) obeying the spherical geometry.
+
+        Parameters
+        ----------
+        RA : float
+            Right ascension of the reference point in degrees.
+        DEC : float
+            Declination of the reference point in degrees.
+        r : float
+            Maximum separation of data points from the reference point.
+
+        Returns
+        -------
+        idx : array_like
+            Positional indices of matching data points in the search tree data
+            with sepration < r.
+        dist : array_like
+            Angular separation of matching data points from reference point.
+        """
+        point_sphere = self._position_sky2sphere(RA, DEC)
+        # find all points that lie within r
+        r_sphere = self._distance_sky2sphere(r)
+        idx = self.tree.query_ball_point(point_sphere, r_sphere)
+        # compute pair separation
+        dist_sphere = minkowski_distance(self.tree.data[idx], point_sphere)
```

Comment:
Wouldn't it be simpler here to just compute the dot product between the found pairs and reference point and then compute arccos? Might be more efficient than computing the distance and having to do the two or there more calculations. It would at least save a few lines of code.

Reply:
After testing here a bit, it doesn't seem to be much if at all slower so keep it for now.